src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/separation/GhostWellDef.scala

```diff
@@ -44,6 +40,12 @@ trait GhostWellDef { this: TypeInfoImpl =>
       case None => noMessages
     }
 
+    case n: PTypeDecl =>
+      error(n, s"ghost error: expected an actual type but found ${n.right}", isTypeGhost(n.right) && !isEnclosingGhost(n)) ++
+      // to avoid confusion about how equality works for this type declaration, we require that the type declaration
+      // is ghost iff its RHS is a ghost type:
```

Comment:
this sounds good to me, but an alternative idea would be to allow any type for which `==` matches `===` on the rhs when the lhs is ghost. this would allow definitions as shown in src/test/resources/regressions/features/ghost_type/ghost-type-decl-fail01.gobra

Reply:
Nice idea! I don't plan to implement it now but I've added a corresponding comment to the implementation